### PR TITLE
[sw, dif_aon_timer] DIF AON Timer to S1

### PR DESCRIPTION
**Pull request by silvestrst**
- It is not clear how `aon_timer.WKUP_CAUSE` is different to `aon_timer.INTR_STATE` from SW perspective - so there is no API to query the "cause" at this time.

**- AON wiring in hardware is broken, need to hook the smoketest into CI when fixed, see #5629**

### CODE REVIEW

File: sw/device/lib/dif/dif_aon_timer.c
Comment (mcy) [inline sw/device/lib/dif/dif_aon_timer.c:20]: These don't need to be defines.

File: sw/device/lib/dif/dif_aon_timer.c
Comment (mcy) [inline sw/device/lib/dif/dif_aon_timer.c:34]: These should be just one stanza...?

File: sw/device/tests/dif/dif_aon_timer_smoketest.c
Comment (mcy) [inline sw/device/tests/dif/dif_aon_timer_smoketest.c:37]: You can just write CHECK(!is_pending)

File: sw/device/tests/dif/dif_aon_timer_unittest.cc
Comment (mcy) [inline sw/device/tests/dif/dif_aon_timer_unittest.cc:375]: Why not `kVariant + 1`...?

File: sw/device/tests/dif/dif_aon_timer_smoketest.c
Comment (tomroberts-lowrisc) [inline sw/device/tests/dif/dif_aon_timer_smoketest.c:56]: I don't think this is right - is_pending is still set from above. Also the Wkup IRQ never gets acknowledged.

File: sw/device/lib/dif/dif_aon_timer.c
Comment (tomroberts-lowrisc) [inline sw/device/lib/dif/dif_aon_timer.c:296]: This isn't quite right for a W1C register - you just need to set the bits you want. This read modify write will clear all the pending bits as well as the one you wanted to clear

**Review by tomroberts-lowrisc — COMMENTED** [review]
Apologies the HW is a bit broken at the moment - I'll fix ASAP!

A couple of comments in the meantime

**Comment by silvestrst** [comment]
> Apologies the HW is a bit broken at the moment - I'll fix ASAP!
> 
> A couple of comments in the meantime

#5629 - the fix for broken AON wiring.

File: sw/device/tests/dif/dif_aon_timer_unittest.cc
Comment (silvestrst) [reply sw/device/tests/dif/dif_aon_timer_unittest.cc:375]: C++ doesn't seem to let you do it that way

File: sw/device/tests/dif/dif_aon_timer_smoketest.c
Comment (silvestrst) [reply sw/device/tests/dif/dif_aon_timer_smoketest.c:56]: Nice catch!

Done.

**Review by silvestrst — COMMENTED** [review]
@mcy and @tomroberts-lowrisc thank you for your reviews!

I have addressed all of your comments.

I have split out `test_main` body into two test functions - one for wake-up and another for watchdog. I think it makes it little bit more readable.

**Please note that the CI change will fail until Tom's HW fixup is merged**

File: sw/device/tests/dif/dif_aon_timer_unittest.cc
Comment (mcy) [reply sw/device/tests/dif/dif_aon_timer_unittest.cc:375]: `int valid = static_cast<int>(...) + 1;`.

Just a matter of knowing the language. Let's try to avoid doing things that might confuse a reader like this that rely on implicit conversion.

**Review by mcy — APPROVED** [review]

**Comment by mcy** [comment]
I left one more comment somewhere in the chain, but LG otherwise.

File: sw/device/lib/dif/dif_aon_timer.h
Comment (mundaym) [inline sw/device/lib/dif/dif_aon_timer.h:121]: Is it worth adding a newline after `/**`while we are here?

**Review by mundaym — APPROVED** [review]
LGTM once mcy's remaining comments are addressed.

File: sw/device/tests/dif/dif_aon_timer_unittest.cc
Comment (silvestrst) [reply sw/device/tests/dif/dif_aon_timer_unittest.cc:375]: The way I done it is just a different way of doing it, but I agree your suggestion is better.

Comment (silvestrst) [reply sw/device/tests/dif/dif_aon_timer_unittest.cc:375]: Done.

File: sw/device/lib/dif/dif_aon_timer.h
Comment (silvestrst) [reply sw/device/lib/dif/dif_aon_timer.h:121]: Ooops, thanks.

Done.

File: sw/device/lib/dif/dif_aon_timer.c
Comment (silvestrst) [reply sw/device/lib/dif/dif_aon_timer.c:89]: Don't think there's much difference, but I have changed it anyway.

Done.

File: sw/device/tests/dif/dif_aon_timer_unittest.cc
Comment (alphan) [inline sw/device/tests/dif/dif_aon_timer_unittest.cc:470]: nit: this and the next comment should not be here.

**Review by alphan — COMMENTED** [review]
Thanks @silvestrst and apologies for the late review. Just some nits.

File: sw/device/lib/dif/dif_aon_timer.c
Comment (silvestrst) [reply sw/device/lib/dif/dif_aon_timer.c:27]: Done.

**Pull request merged**
